src/zope/pluggableauth/plugins/idpicker.py

```diff
@@ -53,11 +59,11 @@ class IdPicker(NameChooser):
     """
     def chooseName(self, name, object):
         i = 0
-        name = unicode(name)
+        name = text_type(name)
         orig = name
         while (not name) or (name in self.context):
             i += 1
-            name = orig+str(i)
+            name = orig + str(i)
```

Comment:
It's not going to break mostly through luck 😄 

On Python 3, `text_type` is `str` and `str` + `str` works out as expected.
```pycon
>>> u'abc' + str(123)
'abc123'
```

On Python 2, `text_type` is `unicode`. `unicode` + `str` does unicode-promotion on the `str` argument; the `str` of an int is only going to have ASCII characters and so will always be promotable to unicode.
```pycon
>>> u'abc' + str(123)
u'abc123'
```

FWIW, `zope.container.contained.NameChooser` (from which this class derives) also makes the exact same assumptions in its `chooseName` method, freely combining `unicode` and `str`:
```python
        # convert to unicode and remove characters that checkName does not allow
        if isinstance(name, bytes):
            name = name.decode()
        if not isinstance(name, unicode):
            try:
                name = unicode(name)
            except:
                name = u''
        name = name.replace('/', '-').lstrip('+@')

        ...
        # for an existing name, append a number.
        # We should keep client's os.path.extsep (not ours), we assume it's '.'
        dot = name.rfind('.')
        if dot >= 0:
            suffix = name[dot:]
            name = name[:dot]
        else:
            suffix = ''

        n = name + suffix
        i = 1
        while n in container:
            i += 1
            n = name + u'-' + str(i) + suffix
```